parser/patches/expensify-common+1.0.0.patch

```diff
@@ -51,82 +43,34 @@ index 1d3d775..6ab5fa3 100644
                  replacement: '<mention-report>$1</mention-report>',
              },
  
-@@ -280,7 +280,7 @@ export default class ExpensiMark {
-                  * Use [\s\S]* instead of .* to match newline
-                  */
-                 name: 'italic',
--                regex: /(\b_+|\b)(?!_blank")_((?![\s_])[\s\S]*?[^\s_])_(?![^\W_])(?![^<]*(<\/pre>|<\/code>|<\/a>|<\/mention-user>|_blank))/g,
-+                regex: /(?<!<[^>]*)(\b_+|\b)(?!_blank")_((?![\s_])[\s\S]*?[^\s_](?<!\s))_(?![^\W_])(?![^<]*>)(?![^<]*(<\/pre>|<\/code>|<\/a>|<\/mention-user>|_blank))/g,
- 
-                 // We want to add extraLeadingUnderscores back before the <em> tag unless textWithinUnderscores starts with valid email
-                 replacement: (match, extraLeadingUnderscores, textWithinUnderscores) => {
-@@ -311,12 +311,12 @@ export default class ExpensiMark {
-                 // \B will match everything that \b doesn't, so it works
-                 // for * and ~: https://www.rexegg.com/regex-boundaries.html#notb
-                 name: 'bold',
--                regex: /\B\*(?![^<]*(?:<\/pre>|<\/code>|<\/a>))((?![\s*])[\s\S]*?[^\s*])\*\B(?![^<]*(<\/pre>|<\/code>|<\/a>))/g,
-+                regex: /(?<!<[^>]*)\B\*((?![\s*])[\s\S]*?[^\s*](?<!\s))\*\B(?![^<]*>)(?![^<]*(<\/pre>|<\/code>|<\/a>))/g,
-                 replacement: (match, g1) => (g1.includes('</pre>') || this.containsNonPairTag(g1) ? match : `<strong>${g1}</strong>`),
-             },
-             {
-                 name: 'strikethrough',
--                regex: /\B~((?![\s~])[\s\S]*?[^\s~])~\B(?![^<]*(<\/pre>|<\/code>|<\/a>))/g,
-+                regex: /(?<!<[^>]*)\B~((?![\s~])[\s\S]*?[^\s~](?<!\s))~\B(?![^<]*>)(?![^<]*(<\/pre>|<\/code>|<\/a>))/g,
-                 replacement: (match, g1) => (g1.includes('</pre>') || this.containsNonPairTag(g1) ? match : `<del>${g1}</del>`),
-             },
-             {
-@@ -446,7 +446,13 @@ export default class ExpensiMark {
-             {
-                 name: 'image',
-                 regex: /<img[^><]*src\s*=\s*(['"])(.*?)\1(?:[^><]*alt\s*=\s*(['"])(.*?)\3)?[^><]*>*(?![^<][\s\S]*?(<\/pre>|<\/code>))/gi,
--                replacement: (match, g1, g2, g3, g4) => `![${g4 || g2}](${g2})`
-+                replacement: (match, g1, g2, g3, g4) => {
-+                    if (g4) {
-+                        return `![${g4}](${g2})`;
-+                    }
-+
-+                    return `!(${g2})`;
-+                }
-             }
-         ];
- 
-@@ -521,7 +527,7 @@ export default class ExpensiMark {
+@@ -579,7 +579,7 @@ export default class ExpensiMark {
           * @type {Number}
           */
          this.currentQuoteDepth = 0;
 -    }
++    },
+ 
+     getHtmlRuleset(filterRules, disabledRules, shouldKeepRawInput) {
```

Comment:
Do we need this function? Where is it used?

Reply:
it's used inside `replace` function of ExpensiMark. It's just not present inside patch file